core/src/test/scala/mill/main/ParseArgsTest.scala

```diff
@@ -0,0 +1,290 @@
+            val Right(out) = ParseArgs.expandBraces(in)
+            assert(
+              out == expectedOut
+            )
+        }
+      }
+    }
+
+    'apply - {
+      'rejectEmpty {
+        val Left(error) = ParseArgs(Seq.empty)
+        assert(error == "Selector cannot be empty")
+      }
+      'singleSelector - {
+        val Right((selectors, args)) =
+          ParseArgs(singleSelector)
+        assert(
+          selectors == List(
+            List(
+              Mirror.Segment.Label("core"),
+              Mirror.Segment.Label("compile")
+            )
+          ),
+          args.isEmpty
+        )
+      }
+      'singleSelectorWithArgs - {
+        val Right((selectors, args)) =
+          ParseArgs(singleSelectorWithArgs)
+        assert(
+          selectors == List(
+            List(
+              Mirror.Segment.Label("application"),
+              Mirror.Segment.Label("run")
+            )
+          ),
+          args == Seq("hello", "world")
+        )
+      }
+      'singleSelectorWithCross - {
+        val Right((selectors, args)) =
+          ParseArgs(singleSelectorWithCross)
+        assert(
+          selectors == List(
+            List(
+              Mirror.Segment.Label("bridges"),
+              Mirror.Segment.Cross(Seq("2.12", "jvm")),
+              Mirror.Segment.Label("compile")
+            )
+          ),
+          args.isEmpty
+        )
+      }
+      'multiSelectorsBraceExpansion - {
+        val Right((selectors, args)) =
+          ParseArgs(multiSelectorsBraceExpansion)
+
+        assert(
+          selectors == List(
+            List(
+              Mirror.Segment.Label("core"),
+              Mirror.Segment.Label("compile")
+            ),
+            List(
+              Mirror.Segment.Label("application"),
+              Mirror.Segment.Label("compile")
+            )
+          ),
+          args.isEmpty
+        )
+      }
+      'multiSelectorsBraceExpansionWithArgs - {
+        val Right((selectors, args)) =
+          ParseArgs(multiSelectorsBraceExpansionWithArgs)
+
+        assert(
+          selectors == List(
+            List(
+              Mirror.Segment.Label("core"),
+              Mirror.Segment.Label("run")
+            ),
+            List(
+              Mirror.Segment.Label("application"),
+              Mirror.Segment.Label("run")
+            )
+          ),
+          args == Seq("hello", "world")
+        )
+      }
+      'multiSelectorsBraceExpansionWithCross - {
+        val Right((selectors, args)) =
+          ParseArgs(multiSelectorsBraceExpansionWithCross)
+
+        assert(
+          selectors == List(
+            List(
+              Mirror.Segment.Label("bridges"),
+              Mirror.Segment.Cross(Seq("2.12", "jvm")),
+              Mirror.Segment.Label("test")
+            ),
+            List(
+              Mirror.Segment.Label("bridges"),
+              Mirror.Segment.Cross(Seq("2.12", "jvm")),
+              Mirror.Segment.Label("jar")
```

Comment:
For all of these let's`import Mill.Segment.{Label,Cross}` and one-line them to reduce verbosity

```scala
List(Label("bridges"), Label("bridges"), Label("test")),
List(Label("bridges"), Label("bridges"), Label("jvm")),
```